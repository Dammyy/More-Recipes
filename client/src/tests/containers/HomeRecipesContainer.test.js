import React from 'react';
import { shallow } from 'enzyme';
import { HomeRecipesContainer, mapStateToProps } from
  '../../containers/HomeRecipesContainer';

describe('HomeRecipesContainer', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        recipesActions: { getRecipes: jest.fn(), },
        recipes: []
      };
      const component = shallow(<HomeRecipesContainer {...props} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('container functions', () => {
    describe('mapStateToProps', () => {
      it('returns the expected properties', () => {
        const state = {
          getIn: () => ({ toJS: jest.fn() })
        };
        const mstp = mapStateToProps(state);
        expect(mstp).toHaveProperty('recipes');
        expect(mstp).toHaveProperty('firstName');
      });
    });
  });
  describe('Instance methods', () => {
    describe('Home Recipes Container instance methods', () => {
      test('', () => {
        const props = {
          recipesActions: {
            getRecipes: jest.fn()
          },
          recipes: []
        };
        const component = shallow(<HomeRecipesContainer {...props} />);
        component.instance().getRecipes();
        expect(props.recipesActions.getRecipes).toHaveBeenCalled();
      });
    });
  });
});
