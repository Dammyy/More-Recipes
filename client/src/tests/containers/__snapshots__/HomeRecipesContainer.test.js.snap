// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeRecipesContainer Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeRecipesContainer
    recipes={Array []}
    recipesActions={
      Object {
        "getRecipes": [MockFunction] {
          "calls": Array [
            Array [
              1,
            ],
          ],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <RecipesListHome
        recipes={Array []}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "recipes": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <RecipesListHome
          recipes={Array []}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "recipes": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
